WordPress/WordPressApi/WordPressComOAuthClient.m

```diff
@@ -72,36 +92,45 @@ - (void)requestOneTimeCodeWithUsername:(NSString *)username
         @"wpcom_resend_otp": @(YES)
     };
     
-    [self POST:@"token"
-    parameters:parameters
-       success:^(AFHTTPRequestOperation *operation, id responseObject) {
-           if (success) {
-               success();
-           }
-       } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
-           error = [self processError:error forOperation:operation];
-           
-           // SORRY:
-           // SMS Requests will still return WordPressComOAuthErrorNeedsMultifactorCode. In which case,
-           // we should hit the success callback.
-           if (error.code == WordPressComOAuthErrorNeedsMultifactorCode) {
-               if (success) {
-                   success();
-               }
-           } else if (failure) {
-               failure(error);
-           }
-       }];
+    [self.sessionManager POST:@"token"
+                   parameters:parameters
+                      success:^(NSURLSessionDataTask *task, id responseObject) {
+                          if (success) {
+                              success();
+                          }
+                      } failure:^(NSURLSessionDataTask *task, NSError *error) {
+                          if ([task.response isKindOfClass:[NSHTTPURLResponse class]]) {
```

Comment:
If I understand the code correctly, if the response is not a `NSHTTPURLResponse` instance, you don't process the error, and you'd never get a `WordPressComOAuthErrorNeedsMultifactorCode`. So in that case we could bail early and call `failure`. In other words, I miss `guard` 😉